pkg/sfu/receiver.go

```diff
@@ -195,6 +195,7 @@ func NewWebRTCReceiver(
 	w.streamTrackerManager.OnAvailableLayersChanged(w.downTrackLayerChange)
 	w.streamTrackerManager.OnBitrateAvailabilityChanged(w.downTrackBitrateAvailabilityChange)
 	w.streamTrackerManager.OnMaxPublishedLayerChanged(w.downTrackMaxPublishedLayerChange)
+	w.streamTrackerManager.OnMaxTemporalLayerSeenChanged(w.downTrackMaxTemporalLayerSeenChange)
 	w.streamTrackerManager.OnBitrateReport(w.downTrackBitrateReport)
```

Comment:
all of these are bound to the same object, wouldn't it be easier to use an interface? this callback pattern defeats the go language server and makes it a lot harder to traverse code - with interfaces we can use gopls on the interface and find all the implementations (usually only one or one real + one fake) and view call hierarchies that include these functions. with the callback pattern the only way to figure out what the callback points to is to manually track down the handler binding function and look up where it gets called and what is passed to it. and some of these callbacks are like 5+ indirections deep...

Reply:
good idea @paulwe . Will look at it.